### PR TITLE
All packages

### PR DESCRIPTION
**Pull request by MichaelKora**

### CODE REVIEW

File: conda_oci_mirror/oras.py [hunk 1]
```diff
@@ -20,9 +21,20 @@ def pull(self, location, subdir, package_name, media_type):
         return self.run(args)
 
     def push(self, target, tag, layers, config=None):
+        
         layer_opts = [f"{str(layer.file)}:{layer.media_type}" for layer in layers]
         dest = f"{target}:{tag}"
+        
+        #upload_url = f"{dest}"
+        #push_cmd = f"oras push {dest} {layer_opts}"
```
Comment (wolfv) [inline conda_oci_mirror/oras.py:29]: can you remove the comments here?

File: conda_oci_mirror/functions.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            with open(index_path, "w") as write_file:
+                json.dump(json_object, write_file)
+                
+            logging.warning(f"upload the index.json file...")
+            upload_path = channel + "/" + subdir + "/" + index_file["name"] + "/index.json" 
+            
+            #upload_index_json(str(dir_index), remote_loc)
+            now = datetime.now()
+            tag = now.strftime("%d%m%Y%H%M%S")
+            upload_cmd = f"oras push {remote_loc}/{upload_path}:latest {upload_path}:application/json"
```
Comment (wolfv) [inline conda_oci_mirror/functions.py:113]: Can you use the ORAS class here? (in oras.py)

**Comment by wolfv** [comment]
Can you squash the commits to a single one or two?

**Review by wolfv — CHANGES_REQUESTED** [review]

File: conda_oci_mirror/functions.py (hunk 2 above)
Comment (MichaelKora) [reply conda_oci_mirror/functions.py]: done

File: conda_oci_mirror/oras.py (hunk 1 above)
Comment (MichaelKora) [reply conda_oci_mirror/oras.py]: done

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
+repos:
+<<<<<<< HEAD
```
Comment (wolfv) [inline .pre-commit-config.yaml]: rebase problem here.

**Pull request closed without merging**